server/src/print.ts

```diff
@@ -40,6 +42,47 @@ export function typeToString(syn: ast.AnyType): string {
     }
 }
 
+export function parens(s: string): string {
+    return `(${s})`;
+}
+
+// To avoid confusion, we will assume 
+// 1. logical operators && and || have equal precedence
+// 2. bitwise (binary) operators have the equal precedence 
+// so we add extra parens in these cases even if unnecessary
+
+export function create_opmap() {
```

Comment:
Use camelCase instead of snakeCase when writing javascript/typescript, and give the function a type annotation